front/components/customSnackBar.component.tsx

```diff
@@ -1,21 +1,83 @@
 import * as React from "react";
+import { useEffect } from "react";
 import { StyleSheet } from "react-native";
-import { Snackbar } from "react-native-paper";
+import * as Animatable from "react-native-animatable";
 
-import { Colors } from "../constants";
+import { Colors, FontWeight, Margins, Paddings } from "../constants";
+import { SecondaryText } from "./StyledText";
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const CustomSnackBar: React.FC<any> = (props) => {
-  return <Snackbar {...props} style={styles.snackBar} />;
+interface Props {
+  duration: number;
+  visible: boolean;
+  isOnTop?: boolean;
+  backgroundColor?: string;
+  onDismiss: () => void;
+  textColor?: string;
+  text: string;
+}
+
+const CustomSnackbar: React.FC<Props> = ({
+  duration,
+  visible,
+  isOnTop,
+  backgroundColor,
+  onDismiss,
+  textColor,
+  text,
+}) => {
+  useEffect(() => {
+    if (visible) {
```

Comment:
J'écrirais ça avec un leave-early et sans les accolades pour le `setTimeout`.

```suggestion
    if (!visible) return;

    setTimeout(() => onDismiss()), duration);
```

Reply:
Yes bonne remarque